pkg/autodiscovery/providers/clusterchecks.go

```diff
@@ -40,7 +41,7 @@ func NewClusterChecksConfigProvider(cfg config.ConfigurationProviders) (ConfigPr
 
 	c.identifier = config.Datadog.GetString("clc_runner_id")
 	if c.identifier == "" {
-		c.identifier, _ = util.GetHostname()
+		c.identifier, _ = util.GetHostname(context.TODO())
```

Comment:
should we add a `context.Context` arg to `NewClusterChecksConfigProvider()` ?

Reply:
If we add a `context.Context` arg to `NewClusterChecksConfigProvider()`, we must add it to all the `New…ConfigProvider()` functions that are passed to `RegisterProvider()`.

The only place where these `NewClusterChecksConfigProvider()` is called is from [`"cmd/agent/common".setupAutoDiscovery(…)`](https://github.com/DataDog/datadog-agent/blob/4744ea6da4830b98c8feb5a2d491990ddc9923ef/cmd/agent/common/autodiscovery.go#L80) which is called only at startup.
As we have a “real” `context` only in the scope of components monitored by health probe and not in the agent startup sequence, we would have, anyhow an empty `context`.

So, moving the “empty” context creation from `NewClusterChecksConfigProvider` to `setupAutoDiscory` wouldn’t provide any benefit.